nova/scheduler/weights/affinity.py

```diff
@@ -93,4 +119,8 @@ def weight_multiplier(self):
     def _weigh_object(self, host_state, request_spec):
         weight = super(ServerGroupSoftAntiAffinityWeigher, self)._weigh_object(
             host_state, request_spec)
+        if weight < 1:
+            # ignore 0.5 returned for hosts don't contain servers of
```

Comment:
Why do you do that here instead of adding a `_weigh_object` into the soft-affinity-weigher calling super? I don't think the soft-anti-affinity-weigher should care about specialities in the soft-affinity-weigher.